stlab/concurrency/default_executor.hpp

```diff
@@ -332,25 +330,26 @@ class priority_task_system {
             std::vector<notification_queue> queues{_count};
             std::swap(q, queues);
             for (unsigned n = 0; n != _count; ++n) {
-                q[n].set_context(_thread_contexts[n].ready);
+                q[n].set_context(_thread_contexts[n]._ready);
             }
         }
         for (unsigned n = 0; n != _count; ++n) {
-            _thread_contexts[n].thread = std::thread([&, n] { run(n); });
+            _thread_contexts[n]._thread = std::thread([&, n] { run(n); });
         }
     }
 
     ~priority_task_system() {
         _done = true;
         for (auto& context : _thread_contexts)
-            context.ready.notify_all();
+            context._ready.notify_all();
 
         for (auto& context : _thread_contexts)
-            context.thread.join();
+            context._thread.join();
     }
 
     template <std::size_t P, typename F>
     void execute(F&& f) {
+        static_assert(0 <= P < 3, "More than 3 priorities are not known!");
```

Comment:
Will `0 <= P < 3` evaluate as you intend? I'd think it would evaluate `0 <= P` first, which will either be `true` or `false`, then cast to an integral value so it can be used in `N < 3`, which will always be `true` (`0 < 3` or `1 < 3`).